src/util/googleSpreadsheetLayer.js

```diff
@@ -85,34 +85,70 @@ L.SpreadsheetLayer = L.LayerGroup.extend({
 
   _getWorksheetID: function(spreadsheetID, spreadsheetFeedURL) {
     var self = this;
-    return $.getJSON(spreadsheetFeedURL, function(data) {
-      // The worksheetID we want is dependent on which sheet we are looking for
-      var tmpLink = data.feed.entry[self.options.sheetNum].id.$t;
-      var sections = tmpLink.split('/');
-      // It is always the last section of the URL
-      var sheetID = sections[sections.length - 1];
-      // Set the URL to the final one.
-      self.options.url = 'https://spreadsheets.google.com/feeds/list/' + spreadsheetID + '/' + sheetID + '/public/values?alt=json';
-    });
+    var request = new XMLHttpRequest();
+    request.open('GET', spreadsheetFeedURL, true);
+    request.onload = function() {     
+      if (this.status >= 200 && this.status < 400) {
+        // Success!
+        var data = JSON.parse(this.response);
+        // The worksheetID we want is dependent on which sheet we are looking for
+        var tmpLink = data.feed.entry[self.options.sheetNum].id.$t;
+        var sections = tmpLink.split('/');
+        // It is always the last section of the URL
+        var sheetID = sections[sections.length - 1];
+        // Set the URL to the final one.
+        self.options.url = 'https://spreadsheets.google.com/feeds/list/' + spreadsheetID + '/' + sheetID + '/public/values?alt=json';
+      } else {
+        // We reached our target server, but it returned an error
+        console.log('server error')
+      }
+    };
+    request.onerror = function() {
+      // There was a connection error of some sort
+      console.log('Something went wrong')
+    };
+    request.send();
   },
 
   requestData: function() {
     var self = this;
     (function() {
-      var script = document.createElement('SCRIPT');
-      script.src = 'https://ajax.googleapis.com/ajax/libs/jquery/1.7.1/jquery.min.js';
-      script.type = 'text/javascript';
-      script.onload = function() {
-        var $ = window.jQuery;
-        var ssURL = self.options.url || '';
+      var ssURL = self.options.url || '';
+      var request = new XMLHttpRequest();
+      request.open('GET', ssURL, true);
```

Comment:
Are there any layers that use this? I couldn't find any using Github search. Any idea how I can check if this works @sagarpreet-chadha? Maybe I should look for spreadsheet data to do so but not sure where to get it from.